### ISSUE TITLE
Feature request: Exit code for skipping subsequent steps

### ISSUE THREAD

**Issue by henribru** [woodpecker-ci/woodpecker#294]
Drone has a custom exit code you can use to exit early and skip subsequent steps, see https://github.com/drone/runner-go/blob/0bd0f8fc31c489817572060d17c6e24aaa487470/pipeline/runtime/execer.go#L277. This is quite useful to avoid running a pipeline if no relevant files changed and it would be awesome to have it in Woodpecker.

**Comment by anbraten** [comment]
Yeah 🎉. I really liked that feature and it would be awesome to have it in Woodpecker as well. 

@henribru If you want to skip a step based on changed files you should have a look at #174.

**Comment by henribru** [comment]
Thanks for linking me to that, that's perfect! Didn't notice it since it's not documented at https://woodpecker.laszlo.cloud/pipeline/#conditional-step-execution yet (though weirdly it's present at https://github.com/woodpecker-ci/woodpecker/blob/master/docs/docs/pipeline.md#conditional-step-execution so it seems like the docs are in need of a redeploy)

**Comment by anbraten** [comment]
You are right the docs are totally outdated right now. We are already working on a complete update of the docs (#282) which will automatically be deployed in the future. You can find a preview of the new docs with the path condition being described at: https://probable-stop.surge.sh/docs/usage/pipeline-syntax#path
